name: .NET Core

on: 
  push:
    branches: archna
    paths: [apis/poi/**,.github/workflows/workflowarchna.yml]
  pull_request:
    branches: master
    paths: [apis/poi/**,.github/workflows/workflowarchna.yml]

jobs:
  build:
    runs-on: ubuntu-latest
    #continue-on-error: true
    steps:
    - uses: actions/checkout@v2
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 3.1.301
    - name: Install dependencies
      run: dotnet restore apis/poi/web/poi.csproj
    - name: Build
      run: dotnet build --configuration Release --no-restore apis/poi/web/poi.csproj
    - name: Test      
      run: dotnet test --no-restore --verbosity normal apis/poi/tests/UnitTests/UnitTests.csproj 
    - name: Open Github Issue
      if: ${{ failure() }}
      uses: rishabhgupta/git-action-issue@v2
      id: Issue
      with:
          token: ${{ secrets.GITHUB_TOKEN }}
          title: Automated Issue creation
          body: This issue is created because tests have failed
          assignees: 'archnaaju'
    - name: Upload Artifact
      uses: actions/upload-artifact@v1.0.0
      with:
        name: POI_Docker_Image
        path: apis/poi/web/
    - name: Checkout
      uses: actions/checkout@v1
    - name: Download built artifact
      uses: actions/download-artifact@master
      with:
          name: webpack artifacts
          path: public
    - name: Build, Tag, Push
      uses: mattdavis0351/actions/docker-gpr@v1.3.0
      with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          image-name: poi-docker
          build-args: |
           ENV SQL_USER=${{secrets.DB_USER_NAME}}
           SQL_PASSWORD=${{secrets.DB_PASSWORD}}
           SQL_SERVER=${{secrets.DB_SERVER}}
           SQL_DBNAME=${{secrets.DB_NAME}}
           WEB_SERVER_BASE_URI='https://openhackxfa5tiw4poi.azurewebsites.net'
    
